cpu/esp_common/esp-wifi/esp_wifi_netdev.c

```diff
@@ -903,6 +904,62 @@ static wifi_config_t wifi_config_ap = {
 };
 #endif /* (defined(MCU_ESP8266) && !defined(MODULE_ESP_NOW)) || defined(MODULE_ESP_WIFI_AP) */
 
+
+#if (defined(MCU_ESP32) && defined(MODULE_ESP_WIFI_CSI))
+void __attribute__((weak)) esp_wifi_csi_rx_cb(void *ctx, wifi_csi_info_t *info)
+{
+    if (!info || !info->buf) {
+        ESP_WIFI_LOG_ERROR("ERROR: esp_wifi_csi_rx_cb missing info or buf");
+        return;
+    }
+    printf("{");
+    printf("\"mac\": \"" MAC_STR "\",", MAC_STR_ARG(info->mac));
+#ifdef MODULE_ESP_WIFI_CSI_METADATA
+    printf("\"rssi\": %d,", info->rx_ctrl.rssi);
+    printf("\"rate\": %d,", info->rx_ctrl.rate);
+    printf("\"sig_mode\": %d,", info->rx_ctrl.sig_mode);
+    printf("\"mcs\": %d,", info->rx_ctrl.mcs);
+    printf("\"cwb\": %d,", info->rx_ctrl.cwb);
+    printf("\"smoothing\": %d,", info->rx_ctrl.smoothing);
+    printf("\"not_sounding\": %d,", info->rx_ctrl.not_sounding);
+    printf("\"aggregation\": %d,", info->rx_ctrl.aggregation);
+    printf("\"stbc\": %d,", info->rx_ctrl.stbc);
+    printf("\"fec_coding\": %d,", info->rx_ctrl.fec_coding);
+    printf("\"sgi\": %d,", info->rx_ctrl.sgi);
+    printf("\"noise_floor\": %d,", info->rx_ctrl.noise_floor);
+    printf("\"ampdu_cnt\": %d,", info->rx_ctrl.ampdu_cnt);
+    printf("\"channel\": %d,", info->rx_ctrl.channel);
+    printf("\"secondary_channel\": %d,", info->rx_ctrl.secondary_channel);
+    printf("\"timestamp\": %d,", info->rx_ctrl.timestamp);
+    printf("\"ant\": %d,", info->rx_ctrl.ant);
+    printf("\"sig_len\": %d,", info->rx_ctrl.sig_len);
+    printf("\"rx_state\": %d,", info->rx_ctrl.rx_state);
+#endif
+    printf("\"first_word_invalid\": %d,", info->first_word_invalid);
+
+    printf("\"buf\": [");
+    for (int i = 0; i < info->len; i++) {
+        printf(PRIu8, (uint8_t)info->buf[i]);
```

Comment:
`PRIu8` doesn't seem to work. Usually buffer data are printed as hex numbers.
```suggestion
        printf("0x%02x", (uint8_t)info->buf[i]);
```